CMakeLists.txt

```diff
@@ -428,12 +428,14 @@ if (NOT SANITIZE)
     set (CMAKE_POSITION_INDEPENDENT_CODE OFF)
 endif()
 
-if (OS_LINUX AND NOT (ARCH_AARCH64 OR ARCH_S390X) AND NOT SANITIZE)
+if (NOT OS_ANDROID AND OS_LINUX AND NOT ARCH_S390X AND NOT SANITIZE)
     # Slightly more efficient code can be generated
-    # It's disabled for ARM because otherwise ClickHouse cannot run on Android.
```

Comment:
the comment suggests that it is intended and thus not a bug

Reply:
That line was added here https://github.com/ClickHouse/ClickHouse/pull/11894 - during some experiments with Android build (at the time when arm support was in very early stages, was not supported by ci/cd etc). The actual work on support of Linux on Arm64 was happening later here: https://github.com/ClickHouse/ClickHouse/issues/24371

I believe that no-pie was just missed then. 

So it's just wrong, it have visible impact on the (advanced) users & developers (can't demangle / symbolize stacktraces after restarting clickhouse), and performance impact. Debugging problems on ClickHouse on [Gravitons](https://www.twitch.tv/videos/2152236905) is problematic w/o that. Because of that i consider it as a bug. 

But if you prefer to consider it 'improvement' / 'whatever' - i'm also ok, and don't care.